### PR TITLE
Add system wide installation option to build script

### PR DESCRIPTION
**Pull request by da-phil**
Because I find the build script very helpful but found it very limiting to be left with installation RT to `$HOME/programs/rawtherapee` I added an option `-s` to the build script to be able to install RT system-wide to `/opt/rawtherapee`.
I also did some minor clean-up, please let me know what you think about it ;)

### CODE REVIEW

**Comment by da-phil** [comment]
@heckflosse, @Desmis, @Beep6581 
Anybody interested in this feature?

**Comment by heckflosse** [comment]
@da-phil 
> Anybody interested in this feature?

I'm on Win10 and don't use the script. I will let others decide on this.

Also ping @Floessie

**Comment by Desmis** [comment]
@da-phil 

Hello

As, Ingo, I don't use the script. Sorry :)

Jacques

**Comment by Floessie** [comment]
I don't use the script either. But I see no problem depending on `sudo` to install RT to `/opt/rawtherapee*`.

File: tools/build-rawtherapee [hunk 1]
```diff
@@ -102,22 +139,19 @@ fi
 
 # Quit if no updates and build-only flag not set
 if [[ "$cloned" = "false" && "$buildOnly" = "false" && "$updates" = "false" && "$existsExe" = "true" ]]; then
-    printf '%s\n' "No updates, nothing to do."
+    printf '%s\n' "No ~, nothing to do."
```
Comment (Floessie) [inline tools/build-rawtherapee]: Hm, what's the intended meaning of `~` here?

Comment (da-phil) [reply tools/build-rawtherapee]: I don't know, looks like a result of a failed search&replace operation :sweat_smile: 
I removed the change in that line.

Can you please check if all is fine now?

**Comment by heckflosse** [comment]
> But I see no problem depending on sudo to install RT

I don't know wheter this script worked on Windows in past. But I know, that there is no sudo in Windows/msys2. At least not by default.

**Comment by da-phil** [comment]
> > But I see no problem depending on sudo to install RT
> 
> I don't know wheter this script worked on Windows in past. But I know, that there is no sudo in Windows/msys2. At least not by default.

I'm also not sure about that as I have no bash experience in Windows, but I would suppose it never would have worked due to all the Unix/Linux specific paths (`$HOME/programs`, `/opt`) and the invocation of `nproc` to find out the number of cores / threads.

Wouldn't it be nice though to have one script to build RT on all supported platforms? ;)
I wonder how people are building and packaging RT currently? manual labor in the terminal? I somehow doubt it :smile:

**Comment by Beep6581** [comment]
Hey

The build script provides a fast, safe and minimal way to compile RawTherapee. The purpose never was to install a compiled program system-wide and the script should not be run with sudo. The same build script is used by numerous other projects and changes must be accepted and propagated accordingly. I don't want to add these changes.

P.S. When editing an existing file, you should preserve the formatting - tabulation, variable capitalization (local script variables should not be capitalized), etc.

**Comment by da-phil** [comment]
> P.S. When editing an existing file, you should preserve the formatting - tabulation, variable capitalization (local script variables should not be capitalized), etc.

I could easily adjust it if that's an issue, system-wide installation is just an optional action and does not hurt if the script is used as before. But just closing the PR? :(

**Comment by Thanatomanic** [comment]
> The same build script is used by numerous other projects and changes must be accepted and propagated accordingly. I don't want to add these changes.

@Beep6581 Numerous seems to be three: RawTherapee, hdrmerge and OpenXCom as far as a search in GitHub tells me. But regardless, why would any of those projects restrict us from making a change to our script, if we agree that the change is useful?

**Comment by Beep6581** [comment]
@Thanatomanic numerous means at least 12 projects.

As stated above, the goal of this script is to have a simple and safe way of building a given program in an isolated space for dev purposes. It's not meant to install the program system-wide, nor to make .desktop file shortcuts or anything else related to installation. These changes are partially blindly copied from the internet, they don't conform to the ideology nor the style of the script, and I have no interest in maintaining them. You're welcome to make these changes to a script under a new name, but please leave this one alone.

**Comment by da-phil** [comment]
> > The same build script is used by numerous other projects and changes must be accepted and propagated accordingly. I don't want to add these changes.
> 
> @Beep6581 Numerous seems to be three: RawTherapee, hdrmerge and OpenXCom as far as a search in GitHub tells me. But regardless, why would any of those projects restrict us from making a change to our script, if we agree that the change is useful?

I don't even understand what this script has to do with the other projects, @Beep6581 can you elaborate?
Otherwise I'm with @Thanatomanic about why we should restrict ourselves in a script used by RT by other projects?


> As stated above, the goal of this script is to have a simple and safe way of building a given program in an isolated space for dev purposes. It's not meant to install the program system-wide, nor to make .desktop file shortcuts or anything else related to installation. These changes are partially blindly copied from the internet, they don't conform to the ideology nor the style of the script, and I have no interest in maintaining them. You're welcome to make these changes to a script under a new name, but please leave this one alone.

If you have carefully looked at my changes you'd see that nothing has changed for script behavior, except there's a new optional option `-s` which can be used to alter the (random) installation path `$HOME/programs` to the usual system-wide installation path. But calling the script as before (`build-rawtherapee -b`, `build-rawtherapee -d`, `build-rawtherapee -b -d`) will give the same result as before. So what's the big deal?
I'm really happy to take constructive criticism for the changes I did, but I don't see the point to have two installation scripts for a project doing the same job (except a different install folder) only because it has to be a carbon copy of the installation script of another project.

**Pull request closed without merging**